### ISSUE TITLE
Introduce the new Widgets editor to the world

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#24685]
### Introduction

Currently as part of Gutenberg's phase 2 process there is an [effort](https://github.com/WordPress/gutenberg/projects/27) put into upgrading the current widgets screen to a block based editor. With the latest progress in #24290 the project is in an architecturally stable point.

It is a good moment to remove the experimental cloak and have it available for a larger user base via the Gutenberg plugin. Issues such as the big list in #24530 need more participation and any help is super valuable. The new UI and UX explorations, even while keeping the same general flow to the current screen will make great use from having the editor exposed to real world projects.

A future step is that after the Widgets  editor is out of being just an experiment, it becomes beta software and the project's focus will be on fixing it, including the default workflow it will offer, and therefore move on with replacing the current screens. 

Aside from using blocks as the base of the UX of managing widgets this project allows users to bring blocks into their widget areas without any required opt in from themes. Testing this important feature via the Gutenberg plugin sooner rather than later is essential for it to work properly.

### Actions

- [ ] Remove the experimental flags that hide the Widgets editor
- [ ] Have the Widgets editor show up under Appearance > Widgets( beta ) in WordPress Admin (for people having the Gutenberg plugin installed)
- [ ] Add a notification in the current widgets screen with an invitation to test the new editor
- [ ] Release the Widgets editor as an optional Appearance link in the next Gutenberg release (8.9)
- [ ] Add the option to restore the current widgets screen and to switch between editors in the Classic Editor plugin
- [ ] Release the Widgets editor as a default Appearance > Widgets link in the next Gutenberg release (9.0)

**Comment by youknowriad** [comment]
>  Have the Widgets editor show up under Appearance > Widgets( beta ) in WordPress Admin (for people having the Gutenberg plugin installed)
 > Add a notification in the current widgets screen with an invitation to test the new editor
>  Release the Widgets editor as an optional Appearance link in the next Gutenberg release (8.9)

Gutenberg is already a development version of WordPress and people installing it know that it's used to test feature before landing on Core. I think building this three items for just 2 weeks (a single release) is just a waste of time, we should move ahead directly to #24087 with a way to disable the new screen (theme option in the current PR).

**Comment by draganescu** [comment]
The key difference here @youknowriad is that #24087 proposes an opt-in to display the editor, while this path is to make it opt-out, with the Classic Editor plugin updates.

While I know we'll run into compatibility issues, it is important that this aspect is cleared sooner rather than later: would the new editors be opt-in or opt-out.

Both the Navigation editor and the Widgets editor will have some BC problems, simply b/c there is a new DOM in place. Maybe [some of the problems](https://github.com/WordPress/gutenberg/issues/23739) will be mitigated but the chances are some won't. However, the data is preserved so switching back and forth between the editors is possible.

**Comment by youknowriad** [comment]
> While I know we'll run into compatibility issues, it is important that this aspect is cleared sooner rather than later: would the new editors be opt-in or opt-out.

I agree we need to clear this point for Core but I don't think it impacts how we roll out the plugin in Gutenberg plugin. I think we need to be a bit more aggressive on the plugin to get all the testing and feedback we need in order to make an informed decision for Core.

I still think it should be opt-int for existing sites and opt-in for new installs (if that's possible to implement) because we know that all customizations to the widgets screen are impossible to support but I'd like that to be proved in the plugin.

**Comment by noisysocks** [comment]
> I still think it should be opt-int for existing sites and opt-in for new installs

@youknowriad: Did you mean to say "**opt-out** for new installs" here?

Also, in the Core Editor chat, @draganescu [suggested that Widgets could be be **opt-out**](https://wordpress.slack.com/archives/C02QB2JS7/p1598452966219200), but #24087 implements **opt-in**.

Personally I'd like the new block-based widgets editor to be **opt-out** for everyone or **opt-out** for new installs.

**Comment by draganescu** [comment]
This is not yet solved as @noisysocks reverted #24087 and #24843 is not merged yet.

**Comment by youknowriad** [comment]
> : Did you mean to say "opt-out for new installs" here?

Yeah @noisysocks I corrected my comment to clarify :) but I was talking about Core there for the plugin, it's opt-out for everyone.
